chia/full_node/coin_store.py

```diff
@@ -565,7 +558,6 @@ async def rollback_to_block(self, block_index: int) -> List[CoinRecord]:
                         coin_changes[record.name] = record
 
             await conn.execute("UPDATE coin_record SET spent_index=0 WHERE spent_index>?", (block_index,))
-        self.coins_added_at_height_cache = LRUCache(self.coins_added_at_height_cache.capacity)
```

Comment:
this looks like we already clear the cache on rollback.

Reply:
Right, but we don't clear it on the other UPDATE when we set the spent index. Originally I cleared the cache there, but that would happen all the time and make the cache useless. So I decided to just remove everything. Alternatively, we could remove spent index from what is being returned in coins_added_at_height_cache since it is not going to be consistent with the DB.